types/datasource.d.ts

```diff
@@ -76,28 +90,196 @@ import {IsolationLevel, Transaction} from './transaction-mixin';
+   * 
+   * @param sql 
+   * @param t Start time 
+   */
+  private log(sql: string, t: number): void;
+
+  private _queuedInvocations: number;
+
+  private _operations: Record<string, OperationOptions>;
+
+  operations(): Record<string, OperationOptions>;
+
+  defineOperation(name: string, options: OperationOptions, fn: Function): void;
+  
+  enableRemote(operation: string): void;
+
+  disableRemote(operation: string): void;
+
+  /**
+   * Default global maximum number of event listeners.
+   * 
+   * @remarks
+   * This default can be overriden through
+   * {@link ConnectorSettings.maxOfflineRequests}.
+   */
+  static DEFAULT_MAX_OFFLINE_REQUESTS: number;
+
+  /**
+   * A hash-map of the different relation types.
+   */
+  static relationTypes: Record<string, string>;
 
-  constructor(name: string, settings?: Options, modelBuilder?: ModelBuilder);
+  constructor(name: string, settings?: ConnectorSettings, modelBuilder?: ModelBuilder);
 
-  constructor(settings?: Options, modelBuilder?: ModelBuilder);
+  constructor(settings?: ConnectorSettings, modelBuilder?: ModelBuilder);
 
   constructor(
     connectorModule: Connector,
-    settings?: Options,
+    settings?: ConnectorSettings,
     modelBuilder?: ModelBuilder,
   );
 
+  private setup(dsName: string, settings: ConnectorSettings): void;
+  private setup(settings: ConnectorSettings): void;
+
+  private _setupConnector();
+
+  private mixin<T extends object>(ModelCtor: T): T
+  
+  /**
+   * Get the maximum number of event listeners
+   * 
+   * @remarks
+   * Defaults to {@link DataSource.DEFAULT_MAX_OFFLINE_REQUESTS} if not explicitly
+   * configured in {@link ConnectorSettings.maxOfflineRequests}.
+   */
+  getMaxOfflineRequests(): number;
+
+  // Reason for deprecation is not clear.
+  /**
+   * {@inheritDoc Connector.getTypes}
+   * @deprecated
+   */
+  getTypes(): string[];
+
+  /**
+   * Check if the datasource supports the specified types.
+   * @param types Type name(s) to check against
+   */
+  supportTypes(types: string | string[]): boolean;
+
+  /**
+   * Returns if the attached connector is relational.
+   * 
+   * @returns If the attached connector is relational; `undefined` if no
+   * connector is attached.
+   */
+  isRelational(): boolean | undefined;
+
+  freeze(): void;
+
+  /**
+   * Return the table name for the specified `modelName`.
+   * 
+   * @param modelName Target model name
+   * @returns The table name
+   */
+  tableName(modelName: string): string;
+
+  /**
+   * Retrieve the column name for the specified `modelName` and `propertyName`.
+   * 
+   * @param modelName Target model name
+   * @param propertyName Target property name
+   * @returns The column name
+   */
+  columnName(modelName: string, propertyName: string): string;
+
+  /**
+   * Retrieve the column names for the specified `modelName`.
+   * 
+   * @param modelName Target model name
+   * @returns Column names
+   */
+  columnNames(modelName: string): string;
```

Comment:
Is this correct? Plural `columnNames` does not make sense with a single `string` return type.